### PR TITLE
Remove obsolete condition argument from expression_is_true

### LINKED ISSUE #695: Remove obsolete condition argument from expression_is_true
### Describe the feature
In #691, @elyobo pointed out:
> why does `expression_is_true` have condition then? Doesn't `where` make it redundant?

And is quite right! 1.0 is a good place to make this breaking change. 

### Describe alternatives you've considered
Leaving it as-is, since we're coming in hot to the 1.0 release date. Since we're fully removing the `not_null_where` and `unique_where` tests, this is just another similar process.

### Who will this benefit?
- People who enjoy consistent APIs
- People who don't want to get confused by legacy behaviour

### Are you interested in contributing this feature?
Nope, but I'd welcome a community contribution (targeting the `utils-v1` branch)!

### PR DESCRIPTION
**Pull request by miles170**
Closes #695 

This is a:
- [ ] documentation update
- [ ] bug fix with no breaking changes
- [ ] new functionality
- [x] a breaking change

All pull requests from community contributors should target the `main` branch (default).

## Description & motivation

Remove `condition` argument since it can be replaced by `where`.

## Checklist
- [x] This code is associated with an Issue which has been triaged and [accepted for development](https://docs.getdbt.com/docs/contributing/oss-expectations#pull-requests). 
- [x] I have verified that these changes work locally on the following warehouses (Note: it's okay if you do not have access to all warehouses, this helps us understand what has been covered)
    - [ ] BigQuery
    - [x] Postgres
    - [ ] Redshift
    - [ ] Snowflake
- [ ] I followed guidelines to ensure that my changes will work on "non-core" adapters by:
    - [ ] dispatching any new macro(s) so non-core adapters can also use them (e.g. [the `star()` source](https://github.com/dbt-labs/dbt-utils/blob/main/macros/sql/star.sql))
    - [ ] using the `limit_zero()` macro in place of the literal string: `limit 0`
    - [ ] using `dbt.type_*` macros instead of explicit datatypes (e.g. `dbt.type_timestamp()` instead of `TIMESTAMP`
- [ ] I have updated the README.md (if applicable)
- [ ] I have added tests & descriptions to my models (and macros if applicable)
- [x] I have added an entry to CHANGELOG.md

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,10 +17,11 @@
 - Add `not_empty_string` generic test that asserts column values are not an empty string. ([#632](https://github.com/dbt-labs/dbt-utils/issues/632), [#634](https://github.com/dbt-labs/dbt-utils/pull/634))
 
 ## Under the hood
-- Remove deprecated table argument from unpivot ([#671](https://github.com/dbt-labs/dbt-utils/pull/671))
+- Remove deprecated table argument from `unpivot` ([#671](https://github.com/dbt-labs/dbt-utils/pull/671))
 - Delete the deprecated identifier macro ([#672](https://github.com/dbt-labs/dbt-utils/pull/672))
 - Handle deprecations in deduplicate macro ([#673](https://github.com/dbt-labs/dbt-utils/pull/673))
-- Fully remove varargs usage in surrogate_key and safe_add ([#674](https://github.com/dbt-labs/dbt-utils/pull/674))
+- Fully remove varargs usage in `surrogate_key` and `safe_add` ([#674](https://github.com/dbt-labs/dbt-utils/pull/674))
+- Remove obsolete condition argument from `expression_is_true` ([#699](https://github.com/dbt-labs/dbt-utils/pull/699))
```
Comment (joellabes) [inline CHANGELOG.md:24]: Can you add some context on what people should do instead? i.e. add a where config

**Review by joellabes — CHANGES_REQUESTED** [review]
Looks great! Could I just get you to flesh out the changelog a smidge more? This one is going to be a reasonably severe breaking change so it'd be good to call out what they should do about it.

File: README.md [hunk 2]
```diff
@@ -156,7 +156,7 @@ models:
           expression: "col_a + col_b = total"
 ```
 
-The macro accepts an optional argument `condition` that allows for asserting
+The macro accepts an optional argument `where` that allows for asserting
```
Comment (joellabes) [inline README.md:159]:
```suggestion
```
This is true for all tests, so I think calling it out here is probably more confusing than is necessary.

File: macros/generic_tests/expression_is_true.sql [hunk 3]
```diff
@@ -1,20 +1,16 @@
-{% test expression_is_true(model, expression, column_name=None, condition='1=1') %}
+{% test expression_is_true(model, expression, column_name=None) %}
 {# T-SQL has no boolean data type so we use 1=1 which returns TRUE #}
 {# ref https://stackoverflow.com/a/7170753/3842610 #}
```
Comment (joellabes) [inline macros/generic_tests/expression_is_true.sql:3]: Don't need this comment anymore

**Review by joellabes — CHANGES_REQUESTED** [review]
Thanks for the updated docs - I just found a couple other nitpicks on the documentation/comments but then this is ready to rock 🎸
